### PR TITLE
Add Mailing-Lists Feature

### PR DESCRIPTION
**Pull request by tanneberger**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,12 +11,16 @@ tower-http = { version = "0.4", default-features = false, features = ["cors", "f
...
+serde_json = { version = "1.0" }
 clap = { version = "4.3", features = ["derive", "env"] }
 url = { version = "2.4", default-features = false }
 markdown = "1.0.0-alpha.11"
 tracing = "0.1"
 anyhow = "1.0"
+email_address = "0.2"
 regex = "1.9"
+reqwest = { version = "0.11", features = ["json"] }
+log = "0.4"
```
Comment (MarcelCoding) [inline Cargo.toml]: just use tracing

File: Cargo.toml [hunk 2]
```diff
@@ -11,12 +11,16 @@ tower-http = { version = "0.4", default-features = false, features = ["cors", "f
...
 serde_yaml = { version = "0.9", default-features = false }
+serde_json = { version = "1.0" }
 clap = { version = "4.3", features = ["derive", "env"] }
 url = { version = "2.4", default-features = false }
 markdown = "1.0.0-alpha.11"
 tracing = "0.1"
 anyhow = "1.0"
+email_address = "0.2"
 regex = "1.9"
+reqwest = { version = "0.11", features = ["json"] }
```
Comment (MarcelCoding) [inline Cargo.toml]: use default-features=false and rustls to avoid openssl build dependency

File: src/lists.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+pub(crate) struct Subscriber {
+  name: String,
+  email: String,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub(crate) struct ListmonkCreateSubscriber {
+  email: String,
+  name: String,
+  status: String, // "enabled" or "disabled"
```
Comment (MarcelCoding) [inline src/lists.rs:30]: Enum?

File: src/lists.rs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+impl MailingLists {
+  pub async fn load(
+    url: &Url,
+    user: &str,
+    password_file: &PathBuf,
+    lists: &[i32],
+  ) -> anyhow::Result<MailingLists> {
+    let mut copied_url = url.clone();
+    copied_url.set_path("/api/subscribers");
```
Comment (MarcelCoding) [inline src/lists.rs]: join?

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

File: src/lists.rs (hunk 3 above)
Comment (tanneberger) [reply src/lists.rs:30]: I just need it int the form

**Review by MarcelCoding — APPROVED** [review]

**Pull request merged**